api/predict.py

```diff
@@ -32,7 +48,7 @@ def post(self):
         j = request.get_json()
         seed_text = j['seed_text']
         gen_chars = j['chars'] if 'chars' in j else DEFAULT_CHARS
-        generated_text = self.model_wrapper.predict(seed_text, gen_chars)
+        generated_text = self.model_wrapper._predict(seed_text, gen_chars)
```

Comment:
Are you sure about the change here? It seems to me quite anti-pattern to call a private function of another class.

Reply:
This was the easiest fix given how the parent class is implemented [here](https://github.com/kmh4321/MAX-Framework/blob/image_utils/maxfw/model/model.py). It's not made for two input variables, but im thinking of just replacing it with a dictionary instead

Reply:
I have changed it and now using the right template for `predict`

Reply:
I think what you mean is that this model accepts multiple input. Perhaps @djalova could shed some light on this? What are the semantics of `_pre_process` and `_post_process`? It seems that existing models are massively ignoring these two methods. Is it like the following in this context:

- `_pre_process` takes the raw JSON string/dict as input, returns the part that is relevant for model input.
- `_predict` takes the JSON input and return the predicted result in a dict.
- `_post_process` takes the dict and construct the JSON string output.